internal/translation/searchindex/searchIndexsvc.go

```diff
@@ -0,0 +1,109 @@
+package searchindex
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/atlas"
+)
+
+var (
+	// ErrNotFound means an resource is missing
+	ErrNotFound = fmt.Errorf("not found")
+)
+
+type AtlasSearchIdxService interface {
+	GetIndex(ctx context.Context, projectID, clusterName, indexName, indexID string) (*SearchIndex, error)
+	CreateIndex(ctx context.Context, projectID, clusterName string, index *SearchIndex) (*SearchIndex, error)
+	DeleteIndex(ctx context.Context, projectID, clusterName, indexID string) error
+	UpdateIndex(ctx context.Context, projectID, clusterName string, index *SearchIndex) (*SearchIndex, error)
+}
+
+type ProductionAtlasSearch struct {
+	searchAPI admin.AtlasSearchApi
+}
+
+func NewAtlasDatabaseUsersService(ctx context.Context, provider atlas.Provider, secretRef *types.NamespacedName, log *zap.SugaredLogger) (*ProductionAtlasSearch, error) {
+	client, err := translation.NewVersionedClient(ctx, provider, secretRef, log)
```

Comment:
so each service maintains its own `client` instance? Is this efficient?

Reply:
The other option is to call the `NewProductionAtlasSearch` ... which BTW i think we agreed to rename on other translation layer PRs. I will revisit that and see if we can do away with this function or not.

Reply:
It was not even used in this case, so I removed. I also renamed the implementing type, no production prefix now.